### PR TITLE
devcomquickwins: Add custom domain for images example

### LINKED ISSUE #2383: Document using Workers for a custom domain with Images
### Expected Behavior
It's a very common request for people to want to serve Images on their own domain. This is on the roadmap however, there is a temporary solution with Workers. It would be nice to have this documented somewhere (either in the Images section or the Workers section) so that it can be easily linked.

Code I have shared in the past for this:
```javascript
addEventListener('fetch', (event) => {
  event.respondWith(handleRequest(event.request));
});

// You can find this in the dashboard, it should look something like this: ZWd9g1K7eljCn_KDTu_MWA
const accountHash = '';

async function handleRequest(request) {
  const { pathname } = new URL(request.url);
  // A request to something like cdn.example.com/83eb7b2-5392-4565-b69e-aff66acddd00/public
  // will fetch "https://imagedelivery.net/<accountHash>/83eb7b2-5392-4565-b69e-aff66acddd00/public"
  return fetch(`https://imagedelivery.net/${accountHash}${pathname}`, request);
}
```

It would also allow people to do something like switch the order of variant and account hash.

---

Personally, I think this can fit in either Images or Workers however I'd understand if the Images team would prefer not having this workaround there currently considering it's on the roadmap.

cc: @Albertozhao @signalnerve - do you think it could fit in the Workers examples/tutorials section if it shouldn't go in Images?

### PR DESCRIPTION
**Pull request by lauragift21**
Write an example on how to use workers to serve images from a custom domain.

Fixes: https://github.com/cloudflare/cloudflare-docs/issues/2383

https://devcom-images-workers-exampl.cloudflare-docs-7ou.pages.dev/workers/examples/images-workers/

### CODE REVIEW

**Comment by rita3ko** [comment]
@lauragift21 do you want to try using the code switched for this one and add a TS example too? :)

**Review by eidam — COMMENTED** [review]
lgtm, and it would make a very nice template for cloudlfare/templates repository :)

File: content/workers/examples/images-workers.md [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+		// will fetch "https://imagedelivery.net/<accountHash>/83eb7b2-5392-4565-b69e-aff66acddd00/public"
+
+		return new Response(`https://imagedelivery.net/${accountHash}${pathname}`);
+	}
+};
+```
+{{</tab>}}
+{{</tabs>}}
+
+Another way you can serve images from a custom domain is by using the `cdn-cgi/imagedelivery` prefix path which is used as path to trigger cdn-cgi image proxy.
```
Comment (deadlypants1973) [inline content/workers/examples/images-workers.md]:
```suggestion
Another way you can serve images from a custom domain is by using the `cdn-cgi/imagedelivery` prefix path which is used as path to trigger `cdn-cgi` image proxy.
```

File: content/workers/examples/images-workers.md [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+
+Here's an example, the hostname is a Cloudflare proxied domain under the same account as the Image, followed with the prefix path and the image `<ACCOUNT_HASH>`, `<IMAGE_ID>` and `<VARIANT_NAME>` which can be found in the Images dashboard.
```
Comment (deadlypants1973) [inline content/workers/examples/images-workers.md]:
```suggestion
Below is an example showing the hostname as a Cloudflare proxied domain under the same account as the Image, followed with the prefix path and the image `<ACCOUNT_HASH>`, `<IMAGE_ID>` and `<VARIANT_NAME>` which can be found in the **Images** on the Cloudflare dashboard.
```

File: content/workers/examples/images-workers.md [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+tags:
+  - Images
+  - Workers
+pcx_content_type: configuration
+title: Custom Domain with Images
+weight: 1001
+layout: example
+---
+
+To serve images from a custom domain, you can create Worker like the example below:
```
Comment (deadlypants1973) [inline content/workers/examples/images-workers.md]:
```suggestion
To serve images from a custom domain:

1. Log in to the [Cloudflare dashboard](https://dash.cloudflare.com).
2. Select you account > select **Workers**.
3. Select **Create a service**.
4. In your Workers, select **Quick edit** and paste the following code.
```

File: content/workers/examples/images-workers.md [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+
+{{<tabs labels="js/esm | ts/esm">}}
+{{<tab label="js/esm" default="true">}}
+```js
```
Comment (deadlypants1973) [inline content/workers/examples/images-workers.md:22]: Making sure that this spacing is intentional?  Should there be another space between 16 and 17?

File: content/workers/examples/images-workers.md [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+		// A request to something like cdn.example.com/83eb7b2-5392-4565-b69e-aff66acddd00/public
+		// will fetch "https://imagedelivery.net/<accountHash>/83eb7b2-5392-4565-b69e-aff66acddd00/public"
+
+		return new Response(`https://imagedelivery.net/${accountHash}${pathname}`);
+	}
+};
+```
+{{</tab>}}
+{{<tab label="ts/esm">}}
+```ts
```
Comment (deadlypants1973) [inline content/workers/examples/images-workers.md:39]: Same comment as above.

File: content/workers/examples/images-workers.md [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+		const { pathname } = new URL(request.url);
+
+		// A request to something like cdn.example.com/83eb7b2-5392-4565-b69e-aff66acddd00/public
+		// will fetch "https://imagedelivery.net/<accountHash>/83eb7b2-5392-4565-b69e-aff66acddd00/public"
+
+		return new Response(`https://imagedelivery.net/${accountHash}${pathname}`);
+	}
+};
+```
+{{</tab>}}
```
Comment (deadlypants1973) [inline content/workers/examples/images-workers.md:54]: Same comment as above.

**Review by deadlypants1973 — CHANGES_REQUESTED** [review]

File: content/workers/examples/images-workers.md (hunk 4 above)
Comment (lauragift21) [reply content/workers/examples/images-workers.md:22]: This is fine, adding a space between the tab and the code block doesn't change the way it displays the code block.

**Comment by lauragift21** [comment]
> lgtm, and it would make a very nice template for cloudlfare/templates repository :)

Thanks @eidam, I will add it in when this is merged.

**Review by deadlypants1973 — APPROVED** [review]

**Pull request merged**